frontend/src/use-cases/category/index.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from "react";
+
+import { Header } from "../../common/views/header";
+import { Footer } from "../../common/views/footer";
+import { DataContext } from "../../common/context/DataContext";
+
+import { Container, RestaurantsContainer } from "./styles";
+import { Restaurant } from "./containers/restaurant";
+
+class CategoryScreen extends Component {
+  render() {
+    return (
+      <Container>
+        <Header />
+        <DataContext.Consumer>
+          {categories => this._renderRestaurants(categories)}
```

Comment:
Since both of the functions you give here:

```js
categories => this._renderRestaurants(categories)
```

and 

```js
_renderRestaurants(categories) { }
```

both take `categories` you can replace the entire value with:

```js
<DataContext.Consumer>
	{_renderRestaurants}
</DataContext.Consumer>
```